src/Frame/VectorFieldSettings.h

```diff
@@ -0,0 +1,70 @@
+/* This file is part of the CARTA Image Viewer: https://github.com/CARTAvis/carta-backend
+   Copyright 2018, 2019, 2020, 2021 Academia Sinica Institute of Astronomy and Astrophysics (ASIAA),
+   Associated Universities, Inc. (AUI) and the Inter-University Institute for Data Intensive Astronomy (IDIA)
+   SPDX-License-Identifier: GPL-3.0-or-later
+*/
+
+#ifndef CARTA_BACKEND__FRAME_VECTORFIELDSETTINGS_H_
+#define CARTA_BACKEND__FRAME_VECTORFIELDSETTINGS_H_
+
+#include <cmath>
+
+#include <carta-protobuf/enums.pb.h>
+#include <carta-protobuf/vector_overlay.pb.h>
+
+struct VectorFieldSettings {
```

Comment:
Great! Could you please create VectorFieldSettings.cc just with
```
/* This file is part of the CARTA Image Viewer: https://github.com/CARTAvis/carta-backend
   Copyright 2018, 2019, 2020, 2021 Academia Sinica Institute of Astronomy and Astrophysics (ASIAA),
   Associated Universities, Inc. (AUI) and the Inter-University Institute for Data Intensive Astronomy (IDIA)
   SPDX-License-Identifier: GPL-3.0-or-later
*/

#include "VectorFieldSettings"
```
then add it to CMakeLists.txt. That way the compiler will process it as a standalone unit and show problems more clearly and not inside Frame (See for instance https://github.com/CARTAvis/carta-backend/blob/dev/src/DataStream/Tile.cc). This is something we should probably do with other pieces of the code, for example, CountorSettings in Frame.h should be extracted, but that's for another issue.